market_maker_keeper/dex_api.py

```diff
@@ -85,6 +85,26 @@ def main(self):
             lifecycle.every(1, self.synchronize_orders)
             lifecycle.on_shutdown(self.shutdown)
 
+    def plunge(self):
+        """
+        Method to automatically plunge any pending transactions on keeper startup
+        """
+        pending_txes = get_pending_transactions(self.web3, self.our_address)
+        logging.info(f"There are {len(pending_txes)} pending transactions in the queue")
+        if len(pending_txes) > 0:
+            if not self.is_unlocked():
+                logging.warning(f"{len(pending_txes)} transactions are pending")
+                return
+            for index, tx in enumerate(pending_txes):
+                logging.warning(f"Cancelling {index+1} of {len(pending_txes)} pending transactions")
+                # Note this can raise a "Transaction nonce is too low" error, stopping the service.
+                # This means one of the pending TXes was mined, and the service can be restarted to either resume
+                # plunging or normal operation.
+                tx.cancel(gas_price=self.gas_price)
+
+    def is_unlocked(self) -> bool:
```

Comment:
Seems strange to test whether account is unlocked just for plunging.  I assume if they account was't unlocked, the keeper would raise errors trying to submit orders upon startup.  Is there a read-only mode?

Reply:
Yeah if the account wasn't unlocked there would be errors, but unfortunately the errors are very general and don't indicate the issue lies with the account - figured it would be better to just explicitly check that they are unlocked given the cost is relatively low. Looking into a read-only mode now.